File: longlong-realtime/src/main/scala/utils/HbaseUtil.scala
package utils

import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.hbase.HBaseConfiguration
import org.apache.hadoop.hbase.client.{Connection, ConnectionFactory}

/**
 * @author longlong
 * @create 2020 07 04 19:32
 * @Describe: 该类主要提供hbase的连接
 */
object HbaseUtil {

  val servers = PropertiesUtil.getProperty("hbase.servers")
  val conf: Configuration = HBaseConfiguration.create()
  conf.set("hbase.zookeeper.quorum", servers)

  def getConnection(): Connection = {
    ConnectionFactory.createConnection(conf)
  }

}

File: longlong-display/src/main/java/com/longlong/longlongdisplay/utils/HbaseUtil.java
package com.longlong.longlongdisplay.utils;

import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.hbase.HBaseConfiguration;
import org.apache.hadoop.hbase.client.Connection;
import org.apache.hadoop.hbase.client.ConnectionFactory;

import java.io.IOException;

/**
 * @author longlong
 * @create 2020 07 05 1:18
 * @Describe: 该类主要提供连接hbase
 */
public class HbaseUtil {

    private static Configuration conf = HBaseConfiguration.create();
    public static Connection getConnection() throws IOException {

        Connection conn = ConnectionFactory.createConnection(conf);
        return conn;
    }
}
